lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -263,6 +277,11 @@ Parser.prototype.queueHeaderBlock = function queueHeaderBlock(header,
         return callback(self.error(constants.error.PROTOCOL_ERROR,
                                    'Header list is too large'));
       }
+      
+      if (header.name.toLowerCase() !== header.name) {
```

Comment:
Gosh, this is potentially a serious performance problem. Any way it could be done simpler?

Perhaps, using `/[A-Z]/.test` should be fine here? Could there be any non-ascii characters?